src/libaktualizr/primary/aktualizr.cc

```diff
@@ -68,6 +68,15 @@ void Aktualizr::UptaneCycle() {
 
   uptane_client_->uptaneInstall(download_result.updates);
   uptane_client_->putManifest();
+  if (uptane_client_->hasPendingUpdates()) {
+    // UptaneCycle() won't do anything useful if there is any pending updates that effectively require either reboot
+    // (ostree) or aktualizr restart (fake pack mngr)
+    LOG_INFO << "About to exit aktualizr since there are pending updates that require either reboot or aktualizr "
+                "restart, no sense to loop Uptane Cycle";
```

Comment:
For the record, merely restarting aktualizr won't do anything interesting, it has to be a full reboot. I think logging this a good idea, but you can probably leave out the "no sense to loop Uptane Cycle" bit at the end.

Reply:
OK. (aktualizr restart helps in case of the fake package manager if fake_need_reboot is on and the sentinel file is removed).